superset/utils/decorators.py

```diff
@@ -111,7 +112,9 @@ def wrapped(*args: Any, **kwargs: Any) -> Any:
     return decorate
 
 
-def on_security_exception(self: Any, ex: Exception) -> Response:
+def on_security_exception(self: Any, ex: Exception) -> Union[Response, werkz_Response]:
+    if not g.user or not utils.get_user_id():
```

Comment:
This can probably be 
```suggestion
   if not g.user or not g.user.is_authenticated:
```

This should also fix @sfirke's issue with the public role